src/rule.rs

```diff
@@ -14,7 +18,11 @@ pub struct RuleMeta {
 
 pub trait Rule: Send + Sync {
     fn meta(&self) -> RuleMeta;
-    fn instantiate(self: Arc<Self>, config: &Config) -> Arc<dyn RuleInstance>;
+    fn instantiate(
+        self: Arc<Self>,
+        config: &Config,
+        rule_configuration: &RuleConfiguration,
```

Comment:
This is where the options really get "applied", since this is the `Rule` -> `RuleInstance` method that gets called for each rule at the beginning of each "run" and is conceptually allowing that rule to be initialized as a function of configuration it seemed to be the right place to tell it what its rule-specific configuration was